pnpm/src/main.ts

```diff
@@ -91,13 +92,25 @@ export async function main (inputArgv: string[]): Promise<void> {
     // we don't need the write permission to it. Related issue: #2700
     const globalDirShouldAllowWrite = cmd !== 'root'
     const isDlxCommand = cmd === 'dlx'
-    config = await getConfig(isDlxCommand ? { ...cliOptions, dir: os.homedir() } : cliOptions, {
+    const currentDirConfig = await getConfig(cliOptions, {
```

Comment:
Maybe move all the logic to getConfig and support a new option like `onlyReadAuthSettingsFromCwd`

Reply:
I have create an option named `inheritNonAuthFromDir` and move the logic to `@pnpm/config`: https://github.com/pnpm/pnpm/pull/8207/commits/7a0deedc5a143bfffdae3f66bef48463ecfb9cfb